format/spec.md

```diff
@@ -315,7 +317,7 @@ Struct evolution requires the following rules for default values:
 * The `write-default` must be set when a field is added and may change
 * When a required field is added, both defaults must be set to a non-null value
 * When an optional field is added, the defaults may be null and should be explicitly set
-* When a new field is added to a struct with a default value, updating the struct's default is optional
+* When a field that is a struct type is added, its default may only be null or non-null. Default values for fields must be stored in field metadata.
```

Comment:
We have a bit of a struct-field vs "field field" ambiguity for me here. 
```suggestion
* When a struct field is added, its default may only be null or non-null. Default values for fields within a struct must be stored in that field's metadata.
```

Reply:
It might be just me, but as it stands, this sentence sounds odd:

> its default may only be null or non-null

I read this as: "this assertion may only be true or false". Which is necessarily true :-)

Imo we should define what we mean by `non-null` here. It seems that it denotes a non-null struct instantiated with default values for all its fields. The Go language as the term "zero-value" for that.

Reply:
The first change in the suggestion, adding a "struct field", is what I am trying to avoid. This requirement is stating what happens when a field is added and that field's type is a struct. Using "struct field" is not clear because all field additions are adding a struct field -- that is, a field within a struct. I'll reword to avoid some of the awkward language here.